core/lib/generators/solidus/install/frontend_templates/break_down_solidus_gem.rb

```diff
@@ -0,0 +1,59 @@
+# # This will unpack the solidus gem into its components without calling `bundle install`.
+#
+# # Nothing to do if the `solidus` gem is not there.
+solidus = Bundler.locked_gems.dependencies['solidus'] or return
+
+# Write and remove into and from a Gemfile
+#
+# This custom injector fixes support for path, git and custom sources,
+# which is missing in bundler's upstream injector for a dependency fetched
+# with `Bundler.locked_gems.dependencies`.
+bundler_injector = Class.new(Bundler::Injector) do
+  def build_gem_lines(conservative_versioning)
+    @deps.map do |d|
+      name = d.name.dump
+      is_local = d.source.instance_of?(Bundler::Source::Path)
+      is_git = d.source.instance_of?(Bundler::Source::Git)
+
+      requirement = if is_local
+                      ", path: \"#{d.source.path}\""
+                    elsif is_git
+                      ", git: \"#{d.git}\"".yield_self { |g| d.ref ? g + ", ref: \"#{d.ref}\"" : g }
+                    elsif conservative_versioning
+                      ", \"#{conservative_version(@definition.specs[d.name][0])}\""
+                    else
+                      ", #{d.requirement.as_list.map(&:dump).join(", ")}"
+                    end
+
+      source = ", :source => \"#{d.source.remotes.join(",")}\"" unless is_local || is_git || d.source.nil?
+
+      %(gem #{name}#{requirement}#{source})
+    end.join("\n")
+  end
+end
+
+to_dependency = ->(component) do
+  Bundler::Dependency.new(
+    component,
+    solidus.requirement,
+    "source" => solidus.source,
+    "git" => solidus.source.try(:uri),
+    "ref" => solidus.source.try(:ref),
+  )
+end
+
+bundler_injector.inject(%w[
+  solidus_core
+  solidus_backend
+  solidus_api
+  solidus_sample
+].map(&to_dependency))
+
+bundler_injector.remove(%w[
+  solidus
+])
+
+# `solidus_frontend` is not sourced from rubygems if the solidus gem was not.
+github_solidus_frontend = '--github solidusio/solidus_frontend' unless solidus.source.is_a? Bundler::Source::Rubygems
+bundle_command("add solidus_frontend --skip-install #{github_solidus_frontend}")
```

Comment:
Instead of adding and later removing, why not add only when solidus_frontend is chosen? 🤔

Reply:
This way we have a common baseline that is supposedly equivalent to the original but with exploded list of components, up to the scripts that don't require `solidus_frontend` to remove it.

Reply:
Given that is the component we want to deprecate and that I wouldn't say undoing an action (or definition of action) is a good pattern for expansion, I would still have it only where it's needed. But not a blocker. I'll be fine with whatever you decide 🙂

Reply:
Done, now the `starter_frontend` just adds the gem, and the break-down recipe doesn't do that anymore.